### PR TITLE
GSREN3D-416 - Enedis prod

### PR DESCRIPTION
**Pull request by toni-garcia-camptocamp**
GSREN3D-416 - Enedis prod

### JIRA issue

https://jira.camptocamp.com/browse/GSREN3D-416

### Description

PR BACK : https://github.com/sigrennesmetropole/cooperation_jn_backend/pull/17

- The user click on enedis
- For now we have one error 500 on the website of enedis, one ticket on the support of enedis is open 
- The user is redirect to the website, i simulate that by directly paste the url 
- We get the annual consumption of user in the page redirection-enedis 
- The user is redirect to the manual input with the value set 

https://github.com/sigrennesmetropole/cooperation_jn_solar/assets/121813008/9139dfaf-b5e8-4ccb-8320-b943b9a3c34d

### CODE REVIEW

File: src/services/api-enedis-dataconnect.ts [hunk 1]
```diff
@@ -24,14 +24,22 @@ class ApiEnedisService {
         'Content-Type': 'application/json',
       }
     )
-    return data.prm
+    if (data !== undefined && data.prm !== undefined) {
+      return data.prm
```
Comment (ismailsunni) [inline src/services/api-enedis-dataconnect.ts:28]: I see, the prm is from here

File: src/components/simulation/ElectricityConsumptionManual.vue [hunk 2]
```diff
@@ -22,6 +24,15 @@ function inputIsCompleted() {
     return true
   }
 }
+
+onBeforeMount(() => {
+  if (enedisStore.isEnedisRedirection) {
+    const annual_consumption = Math.round(
+      consumptionAndProductionStore.annualConsumption
+    )
+    consumption.value = annual_consumption.toString()
```
Comment (ismailsunni) [inline src/components/simulation/ElectricityConsumptionManual.vue:33]: Any reason why it is a string? *just curious

File: src/views/RedirectionEnedisView.vue [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+const route = useRoute()
+const consumptionAndProductionStore = useConsumptionAndProductionStore()
+const enedisStore = useEnedisStore()
+const router = useRouter()
+const mapStore = useMapStore()
+
+onBeforeMount(async () => {
+  mapStore.isLoadingMap = true
+  let prm = route.query.usage_points_id
```
[the remark this answers is not in the document]
Comment (toni-garcia-camptocamp) [reply src/views/RedirectionEnedisView.vue]: yes , i  have add one comment. PRM is the name of user id in enedis

File: src/services/api-enedis-dataconnect.ts (hunk 1 above)
Comment (toni-garcia-camptocamp) [reply src/services/api-enedis-dataconnect.ts:28]: PRM is the name of user id in enedis

File: src/components/simulation/ElectricityConsumptionManual.vue (hunk 2 above)
Comment (toni-garcia-camptocamp) [reply src/components/simulation/ElectricityConsumptionManual.vue:33]: There is no specific reason it's true, we would have put it in number

File: src/stores/interactions.ts [hunk 4]
```diff
@@ -49,6 +49,7 @@ export const useInteractionsStore = defineStore('interactions', () => {
 
   return {
     _activeInteractions,
+    _trigger,
```
[the remark this answers is not in the document]
Comment (toni-garcia-camptocamp) [reply src/stores/interactions.ts:52]: Same here, I don't think there is a specific reason. But it's not related to python convention

**Pull request merged**